bazelisk.go

```diff
@@ -105,8 +105,18 @@ type release struct {
 	Prerelease bool   `json:"prerelease"`
 }
 
-func readRemoteFile(url string) ([]byte, error) {
-	res, err := http.Get(url)
+func readRemoteFile(url string, token string) ([]byte, error) {
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, fmt.Errorf("could not create request: %v", err)
+	}
+
+	if token != "" {
+		req.Header.Set("Authorization", "token "+token)
```

Comment:
This logic might be a bit too GitHub specific, if so I'm open to trying to break it up